File: python/SimpleAppSQL.py
"""SimpleAppSQL.py"""
from pyspark.sql import SparkSession

txtFile = "SimpleAppSQL.py"
spark = SparkSession.builder.appName("Simple Application SQL").getOrCreate()
spark.sparkContext.setLogLevel("WARN")
logData = spark.read.text(txtFile).cache()

numSparks = logData.filter(logData.value.contains("spark")).count()

print("Lines with spark: %i" % (numSparks))

spark.stop()

File: scala/src/main/scala/withsql/SimpleAppSQL.scala
package withsql

import org.apache.spark.sql.SparkSession

/**
 * Simple application demonstrating Spark SQL modified from code at:
 *   https://spark.apache.org/docs/latest/quick-start.html
 */
object SimpleAppSQL {
  def main(args: Array[String]): Unit = {
    val txtFile = "src/main/scala/withsql/SimpleAppSQL.scala"
    val spark = SparkSession.builder.appName("Simple Application").master("local[*]").getOrCreate()
    spark.sparkContext.setLogLevel("WARN")

    val logData = spark.read.textFile(txtFile).cache()
    val numVals = logData.filter(line => line.contains("val")).count()
    println(s"Lines with val: $numVals")
    spark.stop()
  }
}
